### PR TITLE
Stopwatch extension (#38 / #182)

### LINKED ISSUE #38: Move StopwatchExtension to pioneer
From the example code in the JUnit 5 documentation - https://github.com/selesy/junit5/blob/issues/343-document-testrule-compatibility/documentation/src/test/java/example/testrule/StopwatchExtension.java

### PR DESCRIPTION
**Pull request by Bukama**
Implementing a stopwatch extension as discussed in #38 and #99. 

Suggested commit message

```
Implement the stopwatch extension (#38 / #182)

This simple stopwatch extension measures the elapsed time of 
each test (which is executed with the extension) and publishes
the the result as a ReportEntry.

closes: #38
PR: #182
```

closes #38
---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

**Comment by Bukama** [comment]
Don't understand why the Travis fails. I merged current master into the branch, run `applySpotlessJava` and a successfull build, commited and created the Pull Request.

**Comment by Michael1993** [comment]
Seen in the [Travis log](https://travis-ci.org/github/junit-pioneer/junit-pioneer/jobs/661165939#L300) (I hope this works). It seems you are missing a `package-info.java` file?

File: src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java
Comment (Michael1993) [inline src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java]: I'm sorry to bother you with this but wouldn't this get overwritten if I run two test files with the `@Stopwatch` annotation?

Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java]: I'm fairly new to the `TestReporter` as is, so I really am just asking for information.

Comment (Bukama) [reply src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java]: I'm not sure as I used the `TestReporter` for the first time, but as I understand it the context is for each test case. But I'm not sure.

**Comment by Bukama** [comment]
> Seen in the [Travis log](https://travis-ci.org/github/junit-pioneer/junit-pioneer/jobs/661165939#L300) (I hope this works). It seems you are missing a `package-info.java` file?
> 
> Seems fair, you _did_ add a new package.

Thanks, never created such a file - but there's always the first time. Run spotless again, several files were changed. Commitet and pushed them. Let's see what Travis makes out of it

**Comment by Bukama** [comment]
Removed the `[WIP]`. Found a note from Marc how I may write a report. As I would need this also for the `@Issue` extension (yeah it comes slowly closer ;) ) I'll open a own PR for that, so this is not getting to large and can be merged (if reviewers are okay with that)

File: src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java
Comment (Bukama) [reply src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java]: @sormuras Can you enlighten us?

**Comment by Bukama** [comment]
Why did never mentioned that I'm lacking documentation. I mean - ME?!?

**Comment by Bukama** [comment]
Blocked by #218 to get rid of the EventRecorderUtils

**Comment by Bukama** [comment]
Adopted the new testkit engine, but still see no sense in this extension 🤷‍♂️ The published entries are nowhere shown, not in the report, not on console, just nowhere. So I thin we can delete this PR and remove the whole issue as it's nonsense.

**Comment by Michael1993** [comment]
Please don't delete this PR yet.

**Comment by Michael1993** [comment]
Okay, I created #255 so you can see that this extension is **not** useless.

**Comment by Bukama** [comment]
Suggested commit message

```
Implement the stopwatch extension (#83 / #182)

This simple stopwatch extension measures the elapsed time of 
each test (which is executed with the extension) and publishes
the the result as a ReportEntry.

closes: #83
PR: #182
```

File: src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java
Comment (Michael1993) [inline src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java:32]: I honestly think either this check is unnecessary or should throw an `ExtensionConfigurationException`.
This extension can be invoked by two things:
- Adding the `@Stopwatch` annotation (which has `@ExtendWith(StopwatchExtension.class)`)
- Adding `@ExtendWith(StopwatchExtension.class)` to your test class

If the explicit registration is not considered bad configuration then this check is unnecessary.
If the explicit registration is considered bad configuration then this check should be the other way around, i.e.:
```
if (!PioneerAnnotationUtils.isAnyAnnotationPresent(context, Stopwatch.class) {
    throw new ExtensionConfigurationException(/* Meaningful message */);
}
storeNowAsLaunchTime(context);
```

File: src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java
Comment (Michael1993) [inline src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java:39]: The same thing here - unnecessary or the other way around.

File: src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java
Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java:32]: An additional note - you could write a test for this. Have a class annotated with `@ExtendWith(StopwatchExtension.class)` and a `@Test` not annotated with `@Stopwatch`. Execute it with the testkit and the result should be a failed test and an `ExtensionConfigurationException`.

**Review by Michael1993 — CHANGES_REQUESTED** [review]
Aside from minor things, I think this is good. Nicely done!

Please merge the `master` branch into your branch.

File: docs/docs-nav.yml
Comment (Bukama) [reply docs/docs-nav.yml]: Better?

Comment (Michael1993) [reply docs/docs-nav.yml]: It's better, I think. But now it's not in sync with the :page_title: attribute of the actual document. 😢

Comment (Bukama) [reply docs/docs-nav.yml]: Docs aligned (and sime typos fixed)

File: docs/stopwatch.adoc
Comment (Michael1993) [inline docs/stopwatch.adoc:8]: Hm, I'm not sure if this does what you think it does. Shouldn't it be: url[\`text\`] instead of \`url[text]\`

**Review by Michael1993 — APPROVED** [review]
LGTM. 👍

File: docs/stopwatch.adoc
Comment (Bukama) [reply docs/stopwatch.adoc:8]: It creates a link in the code format. And my IDEA ASCII DOC plugin tells me it works (the link is correctly clickable in the preview and opens the desired page)

**Pull request merged**
